openedx/core/djangoapps/content_staging/api.py

```diff
@@ -144,6 +146,37 @@ def _save_static_assets_to_user_clipboard(
             log.exception(f"Unable to copy static file {f.name} to clipboard for component {usage_key}")
 
 
+def save_xblock_to_user_clipboard(block: XBlock, user_id: int, version_num: int | None = None) -> UserClipboardData:
+    """
+    Copy an XBlock's OLX to the user's clipboard.
+    """
+    staged_content = _save_xblock_to_staged_content(block, user_id, CLIPBOARD_PURPOSE, version_num)
+    usage_key = block.usage_key
+
+    # Create/update the clipboard entry
+    (clipboard, _created) = _UserClipboard.objects.update_or_create(
+        user_id=user_id,
+        defaults={
+            "content": staged_content,
+            "source_usage_key": usage_key,
+        },
+    )
+
+    return _user_clipboard_model_to_data(clipboard)
+
+
+def stage_xblock_temporarily(
+    block: XBlock, user_id: int, purpose: str = LIBRARY_SYNC_PURPOSE, version_num: int | None = None,
```

Comment:
Let's make `purpose` required here instead of defaulting to LIBRARY_SYNC. I wouldn't want this being used for something else but still having "library sync" as the purpose by accident.